gui/src/app/SPAnalysis/SPAnalysisDataModel.ts

```diff
@@ -42,17 +44,26 @@ export const initialDataModel: SPAnalysisDataModel = {
     samplingOpts: defaultSamplingOpts
 }
 
-export const serializeAnalysis = (data: SPAnalysisDataModel): string => {
-    const intermediary = {
-        ...data, ephemera: undefined }
-    return JSON.stringify(intermediary)
+export const persistStateToEphemera = (data: SPAnalysisDataModel): SPAnalysisDataModel => {
+    const newEphemera = { ...data.ephemera }
+    getStringKnownFileKeys().forEach(k => newEphemera[k] = data[k])
+    return {
+        ...data,
+        ephemera: newEphemera
+    }
 }
 
-export const deserializeAnalysis = (serialized: string): SPAnalysisDataModel => {
-    const intermediary = JSON.parse(serialized)
-    // Not sure if this is strictly necessary
-    intermediary.ephemera = {}
-    const stringFileKeys = Object.values(SPAnalysisKnownFiles).filter((v) => isNaN(Number(v)));
-    stringFileKeys.forEach((k) => intermediary.ephemera[k] = intermediary[k]);
-    return intermediary as SPAnalysisDataModel
+export const getStringKnownFileKeys = () => Object.values(SPAnalysisKnownFiles).filter((v) => isNaN(Number(v)));
```

Comment:
What is this?

Reply:
The enum identifies the actual names of the fields on the object (which are enforced by the type system). This is a way to get the keys out of the enum so that we can access what those field names are. It's used in the serialization code (SPAnalysisSerialization) to iterate over the field names defined by the `SPAnalysisKnownFiles` type.

...we really should change that name since they aren't actually files per se...

Reply:
I guess my question is why is NAN involved. Just a weird JS thing?

Reply:
OH, sorry. Yeah, that's a bit of an artifact--while I don't think it applies in this case, by default Typescript provides a reverse mapping for enums so that you can go from the keys back to the values. I believe this only works for numeric-valued enums (a lot of this code is specifically trying to solve the problem of a reverse mapping doesn't exist when you're using non-numeric enums) but when iterating over the enum members, you often want to filter out numeric values to avoid funny results.

That said, this is relevant for 1) numeric-valued enums and 2) when looping over their keys, so it can probably be removed!

Reply:
(https://blog.logrocket.com/iterate-over-enums-typescript/ for some background, but again this is probably just pleonastic code in this situation)

Reply:
or https://www.typescriptlang.org/play/?target=10&module=1#code/KYDwDg9gTgLgBMAdgVwLZwLIE8CiL0DeAUHHAIJwC8cA5AGYCWUAzjDQDQlwBCVtzwAMYREAEw5cAwnwCsRAL5EiAeQBGAKyEwAdADcAhgBtkwZgApseNAEptdaDn2CAFmbMBra1QB8cYYmYIQ2BtQwgAczMAAwBpOAZmOAASAnd5KOtrJSA for some of the funky behavior you can get with the auto-reverse mapping

Reply:
Thanks for the links

> so it can probably be removed!

Can it?